### PR TITLE
feat: adding feature toggle system implementation

### PR DESCRIPTION
**Pull request by TheEvilDev**
## Overview
Base feature toggle system we can use in AppKit, and supports integration with Evidently, NeoWeblab, and static feature toggles.

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/feature-toggles/src/context/feature-context.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { createContext } from 'react';
+import IFeature from '../models/feature';
+
+export interface IFeatureContext {
+    getFeature(feature: string): Promise<IFeature>;
+}
+
+export default createContext<IFeatureContext>({
+    getFeature: /* istanbul ignore next */ () => Promise.resolve({}),
+});
```
Comment (tracy-french) [inline packages/feature-toggles/src/context/feature-context.tsx:10]: Could you update this change to switch from tabs to spaces?

File: packages/feature-toggles/src/resources/__tests__/StaticFeatureRepository.spec.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+describe('StaticFeatureRepository', () => {
+    it('should respond with value true for available feature', async () => {
+        const features: StaticFeatures = {
+            'test-feature': {
+                value: true,
+                variation: 'T1',
+            },
+        };
+
+        const sut = new StaticFeatureRepository(features);
```
Comment (tracy-french) [inline packages/feature-toggles/src/resources/__tests__/StaticFeatureRepository.spec.ts:11]: What is "sut"?

File: packages/feature-toggles/src/resources/__tests__/EvidentlyFeatureRepository.spec.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import EvidentlyFeatureRepository from '../EvidentlyFeatureRepository';
+
+jest.mock('aws-sdk/clients/evidently');
+
+describe('EvidentlyFeatureRepository', () => {
+    it('should evaluate a feature for a given entity', async () => {
+        const expectedResult = { variation: 'T1', value: 'DummyValue' };
+
+        const evaluateMock = jest.fn(() => Promise.resolve(expectedResult));
+        (Evidently as any).mockImplementation(() => ({
```
Comment (tracy-french) [inline packages/feature-toggles/src/resources/__tests__/EvidentlyFeatureRepository.spec.ts:11]: The change includes many `any`s in the tests. While it's relatively harmless, it's possible to lie in the tests on accident. Is it possible to cast as the correct type?

**Review by tracy-french — CHANGES_REQUESTED** [review]

File: packages/feature-toggles/src/resources/__tests__/StaticFeatureRepository.spec.ts (hunk 2 above)
Comment (mitchlee-amzn) [reply packages/feature-toggles/src/resources/__tests__/StaticFeatureRepository.spec.ts:11]: "System under test", generic test terminology.

Comment (mitchlee-amzn) [reply packages/feature-toggles/src/resources/__tests__/StaticFeatureRepository.spec.ts:11]: Basically it indicates the thing I am testing in this test.

File: packages/feature-toggles/src/context/feature-context.tsx (hunk 1 above)
Comment (mitchlee-amzn) [reply packages/feature-toggles/src/context/feature-context.tsx:10]: Pretty sure this is all 4 space indents? There's no tabs in this file.

Comment (mitchlee-amzn) [reply packages/feature-toggles/src/context/feature-context.tsx:10]: image

Comment (mitchlee-amzn) [reply packages/feature-toggles/src/context/feature-context.tsx:10]: I see the issue, I don't have eslint hooked up correctly in this new package. Will change.

Comment (mitchlee-amzn) [reply packages/feature-toggles/src/context/feature-context.tsx]: Added eslint, and autofixed these.

File: packages/feature-toggles/src/resources/__tests__/EvidentlyFeatureRepository.spec.ts (hunk 3 above)
Comment (mitchlee-amzn) [reply packages/feature-toggles/src/resources/__tests__/EvidentlyFeatureRepository.spec.ts]: I did one better, I updated these and flipped the eslint rule from warning to error.

**Review by tracy-french — APPROVED** [review]
Sweet! This is awesome! Thank you @TheEvilDev!

**Comment by mitchlee-amzn** [comment]
> Sweet! This is awesome! Thank you @TheEvilDev!

That's @mitchlee-amzn to you now sir!

**Pull request merged**